enhancements/sha-bundle-digest.md

```diff
@@ -0,0 +1,163 @@
+
+---
+
+title: SHA-Bundle-Digest
+
+authors:
+
+- "@theishshah"
+
+reviewers:
+
+- TBD
+
+approvers:
+
+- TBD
+
+creation-date: 2022-01-14
+
+last-updated: 2022-01-14
+
+status:implementable
+
+---
+
+  
+
+# Title
+
+  
+
+Add support for SHA-based Digest for Images
+
+  
+
+## Summary
+
+  
+
+Currently, operators generated by operator-sdk will store and retrieve container images based on the image name and the image tag in the bundle. This EP adds support to Operator-SDK such that images can be stored and retrieved in the bundle with an image name and an image digest as an optional replacement for a tag. Image digests are SHA-256 sums that are hashed from the image layer therefore uniquely identify one specific image.
+
+  
+
+## Motivation
+
+  
+
+This set of changes will allow SDK users in disconnected environments to generate bundles with the image digest
+
+  
+
+Users in air gapped environments are currently unable to get images based on tags, and can only access local registries based on the digest
+
+This change is necessary due to non-deterministic behavior with images identified via tags, this issue specifically occurs in the OCP registry due to how the digest is generated with tag name as inputs
+
+  
+
+### Goals
+
+  
+  
+
+-   Give users the ability to the enable digest-based image resolution via a CLI flag when generating bundles.
+    
+-   Allow user to specify additional images to resolve digests via an environment variable.
+    
+-   Add ability to resolve image digests from image tag dynamically using local docker registry.
+    
+-   OPEN QUESTION: Store digest and tag mappings in file so that user-written code can resolve tags to digest.
+    
+
+  
+
+### Non-Goals
+
+  
+
+-   Migrate existing Operator-SDK-based operators from image tags to image digests.
+    
+-   Identify locations in user code that reference images by tag.
+    
+
+  
+  
```

Comment:
```suggestion
### Non-Goals

-   Migrate existing `operator-sdk` based operators from image tags to image digests.
-   Identify locations in user code that reference images by tag.
```

Reply:
Also add a reference to the Drawback mentioned below. That is also a NON goal. You could just reference it as a link 
```
- [See Drawbacks section](#drawbacks)